### ISSUE TITLE
Project sharing rights need to be checked for fonctionnality and security

### ISSUE THREAD

**Issue by Bachibouzouk** [open-plan-tool/gui#184]
Currently the rights are edit and read but maybe we need more rights and/or define well what we mean the user with sharing rights can and cannot do

**Comment by FredericHirschmueller** [comment]
I think it would be great to have three sharing options:

**read:** the user who received the shared project can only read the project, without the possibility to edit any parameters.

**edit:** the user who received the shared project can edit everything in the project. The user who shared the project does not see the changes of the other user in his account.

**collaborate:** the user who received the shared project can edit everything in the project. The changes that are made within the project applies to the project of both users. That means that all the changes that are made by one user will also change the project of the other user. A little warning should be provided when sharing the project through  "collaboration".

**Comment by Bachibouzouk** [comment]
With **read** can the user duplicate or export the project or not at all? Technically the user could anyway write down the parameters down and reset the energy system on their own account (maybe for the timeseries it would be difficult...)

**Comment by FredericHirschmueller** [comment]
I would suggest that it's also not possible to duplicate or export the project. This would make it more save to share a project without the risk of sharing it with too many people. Even though it's still possible to write down the parameters and reset the energy system.

**Comment by mstich0** [comment]
In Bilbao workshop the students tried to work on shared projects. Even with editing right they however were not allowed to restart a simulation/optimization or duplicate an existing scenario. Users with editing right should however have all rights for the project.

**Comment by mstich0** [comment]
There is a second issue we should fix soon: If somebody shares a Project with you, there is no option to delete this project if one does not want to see it in one´s own project folder anymore.

Also I would agree with Frederic with the suggested options though maybe we can rename "edit" "work with a copy"

**read**: "The user who receives the shared project can only read the project, without the possibility to edit anything. Also exporting or copying of the project or the scenarios is not possible for them."

**work with a copy**: "The user who receives the shared project has access to the current state of the project. It is possible for them to copy the project to make changes in it. The user who shared the project does not see the copy and the changes made."

**collaborate**: "The user who receives the shared project has full access to the project. The changes that are made by them within the project apply to both users, as it is not a copy but the same project they are working with."

A little warning should be provided when sharing the project through "collaboration":
"Be aware that any user the project is shared with collaborate rights with, has the right to change or even delete the project. You might want to create a save copy of the project."

When deleting a Project that was shared with **collaborate** the user is asked if they want to delete the Project for all users that are collaborating in this project, or if they want instead just remove themselves from the project.

When implementing be aware of any doubled timeseries issues.

**Comment by paulapreuss** [comment]
I think it makes sense to check that the project sharing rights work as intended. I have a question about this @mstich0: 

> **read**: "The user who receives the shared project can only read the project, without the possibility to edit anything. Also exporting or copying of the project or the scenarios is not possible for them."
> 
> **work with a copy**: "The user who receives the shared project has access to the current state of the project. It is possible for them to copy the project to make changes in it. The user who shared the project does not see the copy and the changes made."

What would be the use case of being able to share a project to read but not allow the user to make a copy? The way I see it, if the user really wanted to recreate the scenario, they could manually re-build it in a new scenario if they have reading rights to the original scenario - then it just seems silly to not let them make a copy directly. In that case only one option would be necessary (read while allowing to copy). Or am I missing something here?

**Comment by mstich0** [comment]
@paulapreuss Yeah fine for me to cut the option **read** only. Though it doesnt hurt to have it, I also cant imagine that there are many cases where people share sensible data with read only just because it would be more effort to reproduce the scenario and get the data. As @FredericHirschmueller brought this up maybe he has a stronger opinion about it.

**Comment by paulapreuss** [comment]
@josihoppe I think this might go together with the other issue about the timeseries not showing for shared projects. Maybe you can investigate and then we can brainstorm a little.
